ddtrace/propagation/http.py

```diff
@@ -73,6 +73,10 @@ def _possible_header(header):
 _POSSIBLE_HTTP_HEADER_TRACESTATE = _possible_header(_HTTP_HEADER_TRACESTATE)
 
 
+# https://www.w3.org/TR/trace-context/#traceparent-header-field-values
+_TRACEPARENT_HEX_REGEX = "^[a-f0-9]{2}-[a-f0-9]{32}-[a-f0-9]{16}-[a-f0-9]{2}$"
```

Comment:
we should pre-compile this, it'll make a _big_ difference.

```suggestion
_TRACEPARENT_HEX_REGEX = re.compile("^[a-f0-9]{2}-[a-f0-9]{32}-[a-f0-9]{16}-[a-f0-9]{2}$")
```

and then below you can do:

```python
if not _TRACEPARENT_HEX_REGEX.match(tp) or ... 
```